#!/usr/bin/env python
PACKAGE = "lidar_lane_detector"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

#Parameter list.

gen.add("ground_filter_enable", bool_t, 0, "ground filter enable", True)

gen.add("ground_height", double_t, 0, "height in meter", -1.0, 0.100, 3)

gen.add("curb_points", int_t, 0, "Estimated number of points on the curb.[pcs]", 5, 1, 30)


#Minimum X - Maximum X
gen.add("min_x", double_t, 0, "Size of the examined area: Min X [m].", 3, -200, 200)
gen.add("max_x", double_t, 0, "Size of the examined area:: Max X [m].", 30, -200, 200)

#Minimum Y - Maximum Y
gen.add("min_y", double_t, 0, "Size of the examined area: Min Y [m].", -10, -200, 200)
gen.add("max_y", double_t, 0, "Size of the examined area: Max Y [m].", 10, -200, 200)

#Minimum Z - Maximum Z
gen.add("min_z", double_t, 0, "Size of the examined area: Min Z [m]. ", -3, -20, 20)
gen.add("max_z", double_t, 0, "Size of the examined area: Max Z [m].", -0.7, -20, 20)



exit(gen.generate(PACKAGE, "lidar_lane_detector", "LidarFilters"))
